panels/examples/complexbin/animal.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/vimeo/dials/panels"
+)
+
+var _ = panels.Must(p.Register("animals", defaultAnimals()))
+
+type animal struct {
+	Type  string // dog, lion, duck
+	Legs  int8
+	Group string // mammal, reptile, marsupial
+}
+
+type animals struct {
+	m map[string]animal
+}
+
+type animalTraits struct {
+	Name string
+}
+
+func (a *animals) Description(scPath []string) string {
+	return "Get a list of all the animals in the directory that match the criteria"
+}
+
+func (a *animals) ShortUsage(scPath []string) string {
+	return strings.Join(scPath, " ")
+}
+
+func (a *animals) LongUsage(scPath []string) string {
+	return strings.Join(scPath, " ") + "\n\tget all the animals"
+}
+
+func (a *animals) DefaultConfig() interface{}      { return &animalTraits{} }
+func (a *animals) SetupParams() panels.SetupParams { return panels.SetupParams{} }
+
+func (a *animals) Run(ctx context.Context, s *panels.Something) error {
+	var sb strings.Builder
+
+	rootDial := s.DialsPath[0]
+	c := rootDial.View().(*animal)
+	an := s.Dials.View().(*animalTraits)
+
+	for n, animal := range a.m {
+		if an.Name != "" && n != an.Name {
+			continue
+		}
+
+		if c.Group != "" && c.Group != animal.Group {
+			continue
+		}
+
+		if c.Legs != 0 && c.Legs != animal.Legs {
+			continue
+		}
+
+		if c.Type != "" && c.Type != animal.Type {
+			continue
+		}
+
+		fmt.Fprintf(&sb, "Name: %s\tType:%s\tLegs: %d\n", n, animal.Type, animal.Legs)
+	}
+	if len(sb.String()) == 0 {
+		fmt.Fprint(&sb, "No animals found matching the specified requirements\n")
+	}
+	fmt.Fprint(s.W, sb.String())
```

Comment:
Hmm, I don't think there's any benefit to using `fmt.Fprint` for a string like this over directly calling `s.W.Write([]byte())`. (or using `io.WriteString()` to do the cast for us)